pkg/controller/deployment/util/deployment_util.go

```diff
@@ -67,6 +67,10 @@ const (
 	RollbackTemplateUnchanged = "DeploymentRollbackTemplateUnchanged"
 	// RollbackDone is the done rollback event reason
 	RollbackDone = "DeploymentRollback"
+	// OverlapAnnotation marks deployments with overlapping selector with other deployments
+	// TODO: Delete this annotation when we no longer need to support a client
```

Comment:
Cleanup is being done in this PR so it will begin in 1.6. Filed #43322 for later removal.